.github/workflows/build-test.yml

```diff
@@ -216,6 +216,130 @@ jobs:
+
+    # This job waits on the full build, even though it doesn't need anything from it.
+    # This way it is only triggered if the build is successful. The CI could be made
+    # a little bit faster by starting this job immediately, as it's fully independent.
+    needs: [build]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Nightly is only needed for the `--profile bench` line. Once that is stabilized,
+      # then this line is no longer needed.
+      - name: Install nightly toolchain
+        run: rustup toolchain install nightly
+
+      - name: Install cargo-make
+        uses: davidB/rust-cargo-make@v1
+        with:
+          version: latest
+
+      - name: Run the example with dhat-rs to collect memory information
+        run: |
+          cargo run --package memory_bench -- ${{ matrix.os }} ${{ matrix.example.component }} ${{ matrix.example.examples }}
+
+      # Benchmarking & dashboards job > (unmerged PR only) Convert benchmark output into
+      # dashboard HTML in a commit of a branch of the local repo.
+
+      - name: Store benchmark result & create dashboard (unmerged PR only)
+        # any action that is not a merge to master implies unfinished PR
+        if: github.event_name != 'push' || github.ref != 'refs/heads/master'
+        uses: gregtatum/github-action-benchmark@v1
+        with:
+          name: Heap – ${{ matrix.example.component }}
+          # The ndjson tool is only supported by the gregtatum fork of github-action-benchmark.
+          tool: 'ndjson'
+          benchmark-data-dir-path: ./benchmarks/memory/${{ matrix.os }}/${{ matrix.example.component }}
+          output-file-path: ./benchmarks/memory/${{ matrix.os }}/${{ matrix.example.component }}/output.ndjson
+          # Show alert with commit comment on detecting possible memory regression
+          alert-threshold: '200%'
+          fail-on-alert: true
+          # Requires one-time-only creation of this branch on remote repo. This will
+          # store the generated information.
+          gh-pages-branch: unmerged-pr-memory-data
+
+          # Do not store historical benchmark info of unfinished PRs. Commits seem to get
+          # made anyways, so make sure that the branch in `gh-pages-branch` is
+          # different from the branch used for merges to master/main branch.
+          auto-push: false
+
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          comment-on-alert: true
+          alert-comment-cc-users: '@sffc,@zbraniecki,@echeran,@gregtatum'
+
+      # Benchmarking & dashboards job > (PR merge to master only) Convert benchmark output
+      # into dashboard HTML in a commit of a branch of the local repo.
+
+      - name: Store benchmark result & create dashboard (merge to master only)
+        # only merges to master (implies PR is finished and approved by this point)
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        uses: gregtatum/github-action-benchmark@v1
```

Comment:
The `@___` syntax in GH Actions specifies a ref. Usually, the refs are tags (named as the semver). If not the semver tag, people usually use the git commit hash. In general, I like versions being immutable, whereas having a branch called `v1` and using it as the ref is not quite that -- can you change `v1` to be the commit hash?

Reply:
Yes, that sounds sensible to me, and I handled it. We should probably do this as well for `rhysd/github-action-benchmark`. This is the recommended practice for hardening security:

https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions

One thing on this new approach is that I'm getting a warning notice which is a bit confusing.

From: https://github.com/gregtatum/icu4x/actions/runs/501955931

> gregtatum/github-action-benchmark@d3f06f738e9612988d575db23fae5ca0008d3d12 looks like the shortened version of a commit SHA. Referencing actions by the short SHA will be disabled soon. Please see https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions. 

However, `d3f06f738e9612988d575db23fae5ca0008d3d12` is the full hash as far as I can tell.